core/src/main/java/com/linecorp/armeria/client/websocket/WebSocketSession.java

```diff
@@ -123,9 +128,29 @@ public void setOutbound(Publisher<? extends WebSocketFrame> outbound) {
             streamMessage = new PublisherBasedStreamMessage<>(outbound);
         }
 
-        if (!outboundFuture.complete(
-                streamMessage.map(webSocketFrame -> HttpData.wrap(encoder.encode(ctx, webSocketFrame))))) {
-            streamMessage.abort();
+        final StreamMessage<HttpData> data =
+                streamMessage.map(webSocketFrame -> HttpData.wrap(encoder.encode(ctx, webSocketFrame)))
+                        .recoverAndResume(cause -> {
+                            if (cause instanceof ClosedStreamException) {
+                                return StreamMessage.aborted(cause);
+                            }
```

Comment:
Question) Do you mind leaving a explanation on why this condition was added?

Reply:
By definition, I thought if a `ClosedStreamException` is passed the corresponding request/response would already be closed